demos/mqtt/mqtt_demo_basic_tls/mqtt_demo_basic_tls.c

```diff
@@ -253,7 +528,23 @@ static SSL * tlsSetup( int tcpSocket )
         SSL_CTX_free( pSslSetup );
     }
 
-    return pSslContext;
+    /* Log failure or success and update the correct exit status to return. */
+    if( status == 0 )
+    {
+        LogError( ( "Failed to establish a TLS connection to %.*s.",
+                    BROKER_ENDPOINT_LENGTH,
+                    BROKER_ENDPOINT ) );
+        status = EXIT_FAILURE;
+    }
+    else
+    {
+        LogInfo( ( "Established a TLS connection to %.*s.\n\n",
+                   BROKER_ENDPOINT_LENGTH,
+                   BROKER_ENDPOINT ) );
+        status = EXIT_SUCCESS;
+    }
```

Comment:
These lines seem odd to me since it's effectively `status = !status`. I think the return value should be a separate variable

Reply:
Yea agree. fixed.